torch/onnx/_internal/fx/function_dispatcher.py

```diff
@@ -3,281 +3,199 @@
+        """
+        aten_op = node.target
+
+        if aten_op == operator.getitem:
+            aten_name = "aten::getitem"
+        elif isinstance(aten_op, torch._ops.OpOverloadPacket):
+            # aten::sym_size is the only OverloadPacket that we support.
+            # schema: aten::sym_size(Tensor self, int dim) -> Tensor
+            if aten_op != torch.ops.aten.sym_size:
+                raise ValueError(
+                    f"Unsupported OverloadPacket: {aten_op}, aten.sym_size is the only allowed OverloadPacket!"
+                )
+            # TODO(titaiwang): aten::sym_size has overload, but fx graph is using
+            # overloadpacket for some reasons.
+            # https://github.com/pytorch/pytorch/issues/97201
+            aten_op_default = aten_op.default
+            aten_name = aten_op_default.name()
+        elif aten_op in _SYMINT_SYMFLOAT_BUILTIN_TO_EXPORTER_KEY_TABLE:
+            # Make sure it's symint/symfloat consuming builtin ops.
+            for node_arg in node.args:
+                if (not isinstance(node_arg, (torch.fx.Node, int, float))) or (
+                    isinstance(node_arg, torch.fx.Node)
+                    and not isinstance(
+                        node_arg.meta["val"], (torch.SymInt, torch.SymFloat)
+                    )
+                ):
+                    raise ValueError(
+                        f"Unsupported node arg: {node_arg} with builtin function: {aten_op},"
+                        " only int/float/SymInt/SymFloat is supported with built-in ops!"
+                    )
+            aten_op = _SYMINT_SYMFLOAT_BUILTIN_TO_EXPORTER_KEY_TABLE[aten_op]
+            aten_name = aten_op.name()
+        elif isinstance(aten_op, torch._ops.OpOverload):
+            aten_name = aten_op.name()
+        else:
+            raise RuntimeError(f"Unknown call_function target: {node.target}")
+
+        func = self.registry.get(aten_name, None)
+        if func is None:
+            # Fall back default if there is no overload
+            if isinstance(aten_op, torch._ops.OpOverload) and hasattr(
+                aten_op.overloadpacket, "default"
+            ):
+                func = self.registry.get(aten_op.overloadpacket.default.name(), None)
+            else:
+                raise RuntimeError(
+                    f"aten name: {aten_name} is not registered in the torchlib registry!"
+                )
+
+        # opschema algo
+        # TODO: Cache the OpSchemaWrapper so we don't need to run the init logic everytime
+        function_opschema = OpSchemaWrapper(func.op_schema)
+        for f in func.overloads:
+            if function_opschema.match_inputs(onnx_args, onnx_kwargs):
+                return f
+            if function_opschema.nearest_match_inputs(onnx_args, onnx_kwargs):
+                return f
+        raise RuntimeError(f"There are no overloaded function for {aten_name}")
+
+
+fx_dispatcher = Dispatcher(registration.default_registry)
```

Comment:
Does `Registry` support cloning?

I'm thinking in the context of when we move to non global dispatcher & registry for `dynamo_export`. Also consider custom op registering where registry is modified.

Reply:
Let me reference on the old one, and see how we can support converter side registration.